src/components/safe-messages/MsgModal/index.tsx

```diff
@@ -102,94 +140,109 @@ const MsgModal = ({
     setSubmitError(undefined)
 
     try {
+      // If you are connected through WC
       if (requestId && !alreadyProposedMessage) {
         await dispatchSafeMsgProposal({ onboard, safe, message: decodedMessage, requestId, safeAppId })
+        setIsPolling(true)
       } else {
         await dispatchSafeMsgConfirmation({ onboard, safe, message: decodedMessage, requestId })
       }
 
-      onClose()
+      if (ongoingMessage && ongoingMessage?.confirmationsSubmitted === ongoingMessage?.confirmationsRequired - 1) {
+        // TODO: send the fully signed message
+        onClose()
+      }
     } catch (e) {
       setSubmitError(e as Error)
     }
-  }, [alreadyProposedMessage, decodedMessage, onClose, requestId, safe, safeAppId, onboard])
+  }, [alreadyProposedMessage, decodedMessage, onClose, requestId, safe, safeAppId, onboard, ongoingMessage])
+
+  const handleClose = () => {
+    if (!alreadyProposedMessage || alreadyProposedMessage?.status === SafeMessageStatus.NEEDS_CONFIRMATION) {
```

Comment:
I think he have to check the ongoingMessage here
```suggestion
    if (!ongoingMessage || ongoingMessage?.status === SafeMessageStatus.NEEDS_CONFIRMATION) {
```